src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,192 @@
+
+alphabet_length(c::StatisticalComplexity) = alphabet_length(c.est)
+
+function complexity(c::StatisticalComplexity, p::Probabilities; h::Float64) 
+    L = alphabet_length(c)
+    # fill with the probability distribution zeros for empty bins, 
+    # we need this to compare to a uniform distribution of m! patterns
+    p = Probabilities(cat(zeros(L-length(p)), p.p, dims=1))
+    # normalization is done automatically by Probabilities
+    uniform = Probabilities(ones(L))
+
+    dist = evaluate(c.distance, p, uniform)
+    dist *= h
+
+    return dist
+end
+
+function complexity(c::StatisticalComplexity, x::AbstractArray)
+    p = probabilities(x, c.est)
+    L = alphabet_length(c)
+    h = entropy(c.entropy, p) / log(c.entropy.base, L)
+    complexity(c, p; h)
+end
+
+function complexity_normalized(c::StatisticalComplexity, p::Probabilities; h::Float64)
+    comp = complexity(c, p; h)
+    L = alphabet_length(c)
+    uniform = Probabilities(ones(L))
+    # generate "deterministic" distribution (only one completely full bin)
+    determ = Probabilities(Base.append!(zeros(L-1), 1.0))
+    # distance to this distribution is the maximum possible distance to a uniform distribution
+    return comp / evaluate(c.distance, determ, uniform)
+end
+
+function complexity_normalized(c::StatisticalComplexity, x::AbstractArray)
+    comp = complexity(c, x)
+    L = alphabet_length(c)
+    uniform = Probabilities(ones(L))
+    # generate "deterministic" distribution (only one completely full bin)
+    determ = Probabilities(Base.append!(zeros(L-1), 1.0))
+    # distance to this distribution is the maximum possible distance to a uniform distribution
+    return comp / evaluate(c.distance, determ, uniform)
+end
+
+linearpermissiverange(start; stop, length) = length==1 ? (start:start) : range(start, stop=stop, length=length)
+
+"""
+    maximum_complexity_entropy(c::StatisticalComplexity; num=1)
+
+Calculate the maximum complexity-entropy curve for the statistical complexity according to [^Rosso2007]
+for `num * m!` different values of the normalized permutation entropy.
+
+## Description
+The way the statistical complexity is designed, there is a minimum and maximum possible complexity
+for data with a given permutation entropy.
+The calculation time of the maximum complexity curve grows as O((m!)^2), and thus takes very long for higher `m`.
+
+This function is adapted from S. Sippels implementation in statcomp [^statcomp].
+
+[^Rosso2007] Rosso, O. A., Larrondo, H. A., Martin, M. T., Plastino, A., & Fuentes, M. A. (2007). 
+            [Distinguishing noise from chaos](https://doi.org/10.1103/PhysRevLett.99.154102).
+            Physical review letters, 99(15), 154102.
+
+[^statcomp] Sippel, S., Lange, H., Gans, F. (2019).
+            [statcomp: Statistical Complexity and Information Measures for Time Series Analysis](https://cran.r-project.org/web/packages/statcomp/index.html)
+"""
+function maximum_complexity_entropy(c::StatisticalComplexity; num::Int=1)
```

Comment:
I suspect this function is not writen with algorithmically optimal performance. These 3-level nested loops hint that things could be done better. But I am not sure how... What I know for sure is that one of the loops can be threaded with `Threads.@threads` because it only writes to the second dimension of a matrix. This is the `k` loop.